### PR TITLE
fix: compute insufficient balance and approval off of input

### PR DESCRIPTION
**Pull request by zzmp**
Improves perceived latency by computing insufficient balance and approval off of input instead of trade.
Improves UX by not updating trade currency amounts until trade is computed, so that old amounts stay (in loading state).

### CODE REVIEW

File: src/lib/hooks/swap/useSwapInfo.tsx [hunk 1]
```diff
@@ -154,6 +152,7 @@ export function SwapInfoUpdater() {
   const setSwapInfo = useUpdateAtom(swapInfoAtom)
   const swapInfo = useComputeSwapInfo()
   useEffect(() => {
+    console.log('zzmp', 'setSwapInfo', swapInfo.currencyAmounts.INPUT)
```
Comment (ianlapham) [inline src/lib/hooks/swap/useSwapInfo.tsx:155]: can delete this

File: src/lib/useInputCurrencyAmount.ts [hunk 2]
```diff
@@ -0,0 +1,16 @@
+import { Currency, CurrencyAmount } from '@uniswap/sdk-core'
+
+import { useIsAmountPopulated, useIsSwapFieldIndependent, useSwapAmount, useSwapCurrency } from './hooks/swap'
+import { Field } from './state/swap'
+import tryParseCurrencyAmount from './utils/tryParseCurrencyAmount'
+
+export default function useInputCurrencyAmount(): CurrencyAmount<Currency> | undefined {
```
Comment (ianlapham) [inline src/lib/useInputCurrencyAmount.ts:7]: why is this in the root level?

**Review by ianlapham — COMMENTED** [review]
A few small comments

**Comment by Rds2008ca-libellule** [comment]
#3318

**Comment by Rds2008ca-libellule** [comment]
@Rds2008ca-libellule

**Review by ianlapham — APPROVED** [review]

**Pull request merged**